src/ctl/service/r2/store/kv/store.go

```diff
@@ -439,6 +439,123 @@ func (s *store) FetchRollupRuleHistory(
 	return nil, rollupRuleNotFoundError(namespaceID, rollupRuleID)
 }
 
+//nolint:gocritic
+func (s *store) FetchUtilizationRule(
+	namespaceID string,
+	utilizationRuleID string,
+) (view.RollupRule, error) {
+	ruleset, err := s.FetchRuleSetSnapshot(namespaceID)
+	if err != nil {
+		return view.RollupRule{}, handleUpstreamError(err)
+	}
+
+	for _, rr := range ruleset.UtilizationRules {
+		if rr.ID == utilizationRuleID {
+			return rr, nil
+		}
+	}
+
+	return view.RollupRule{}, utilizationRuleNotFoundError(namespaceID, utilizationRuleID)
+}
+
+//nolint:gocritic,dupl
```

Comment:
Is `dupl` a check for chunks of copy/paste code? (Unfortunately I think we need to right now).

Reply:
yeah 😞